### ISSUE TITLE
`calicoctl version` returns 0 when connection to the cluster fails

### ISSUE THREAD

**Issue by dmlemos** [projectcalico/calicoctl#1803]
## Expected Behavior

When checking calicoctl version, the return error should be non-zero, if it fails to connect to the cluster:
```sh
$ calicoctl version; echo $?
Client Version:    v1.6.3
Build date:        2017-12-20T22:33:02+0000
Git commit:        d4cfc95c
Cluster Version:   unknown (client: etcd cluster is unavailable or misconfigured; error #0: dial tcp 127.0.0.1:2379: getsockopt: connection refused
)
Cluster Type:      unknown (client: etcd cluster is unavailable or misconfigured; error #0: dial tcp 127.0.0.1:2379: getsockopt: connection refused
1
```

In comparison `kubectl` fails in the same situation:
```sh
$ kubectl version; echo $?
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.2", GitCommit:"5fa2db2bd46ac79e5e00a4e6ed24191080aa463b", GitTreeState:"clean", BuildDate:"2018-01-18T21:12:46Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
The connection to the server localhost:8080 was refused - did you specify the right host or port?
1
```

## Current Behavior

```sh
$ calicoctl version; echo $?
Client Version:    v1.6.3
Build date:        2017-12-20T22:33:02+0000
Git commit:        d4cfc95c
Cluster Version:   unknown (client: etcd cluster is unavailable or misconfigured; error #0: dial tcp 127.0.0.1:2379: getsockopt: connection refused
)
Cluster Type:      unknown (client: etcd cluster is unavailable or misconfigured; error #0: dial tcp 127.0.0.1:2379: getsockopt: connection refused
0
```

Returns 0, even with errors.

## Possible Solution

Have the command fail if it encounters errors.
For now I can use `calicoctl --version` to verify the version. Although this is not documented.

## Context

Validating `calicoctl` server connectivity.

## Your Environment

* calicoctl version (e.g. `calicoctl version`): 1.6.3

**Comment by caseydavenport** [comment]
Yep, this seems like a bug, and should be fairly straightforward to fix - a good one for any new contributors that might be watching :)

**Comment by caseydavenport** [comment]
Looks like this was fixed in v3.0: https://github.com/projectcalico/calicoctl/blob/master/calicoctl/commands/version.go#L70-L74

So this probably just applies to the v2.6 release series, as can be seen here: https://github.com/projectcalico/calicoctl/blob/release-v2.6/calicoctl/commands/version.go#L69-L82

**Comment by caseydavenport** [comment]
This has been fixed in three releases now, so I'm going to close.
